lychee-bin/src/cache.rs

```diff
@@ -0,0 +1,38 @@
+use anyhow::Result;
+use dashmap::DashMap;
+use lychee_lib::{CacheStatus, Uri};
+use std::path::Path;
+
+/// The cache stores previous response codes
+/// for faster checking. At the moment it is backed by `DashMap`, but this is an
+/// implementation detail, which may change in the future.
+pub(crate) type Cache = DashMap<Uri, CacheStatus>;
+
+pub(crate) trait StoreExt {
+    fn store<T: AsRef<Path>>(&self, path: T) -> Result<()>;
```

Comment:
Since this is internal API, we don't need to make it generic (unless it's necessary).
I guess we can use &Path instead.

The potential benefit is faster compilation.

Reply:
True. Although I "benchmarked" it with a few compiles and I couldn't find a difference. Probably because the rest takes longer. 😆